DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs

```diff
@@ -170,7 +170,9 @@ public IEnumerable<GroupDelegate> GetGroupDelegates(int groupId)
         public IEnumerable<GroupCourse> GetGroupCoursesForCentre(int centreId)
         {
             return connection.Query<GroupCourse>(
-                $"{GroupCourseSql} AND c.CentreId = @centreId",
+                @$"{GroupCourseSql}
+                        AND (c.CentreID = @centreId OR c.AllCentres = 1)
+                        AND EXISTS (SELECT CentreApplicationID FROM CentreApplications WHERE (ApplicationID = c.ApplicationID) AND (CentreID = @centreID) AND (Active = 1))",

```

Comment:
I'd argue the new addition would fit better in the GroupCourseSql, and the other methods that use it should be checked and refactored, but that might be unnecessary. Thoughts?

If keeping this new addition where it is, it would be nice to split it up so it it is not just one big line.

Reply:
I've modified this so that it is always included in the SQL, which made a method in the data service redundant so I've removed it, and I think I've caught all the places it is used on the ticket.